src/Adapter/AbstractAdapter.php

```diff
@@ -4,5 +4,21 @@
 
 class AbstractAdapter implements AdapterInterface
 {
+    protected $trainingData;
+
+    public function getTrainingData()
+    {
+        return $this->trainingData;
+    }
+
+    public function train()
+    {
+        // TODO: Implement train() method.
+    }
+
+    public function predicte()
```

Comment:
oki for fisrt comment, 
i tought of having two separate adapters as you said, will do that.
I'm not fond of the call and exec methods, i understand the need to have a unified Interface, but call and exec have no meaning, whilst train, makePreditcion, classify, are easier to remember and read.
What do you think?